File: 1st year/hotel project java/project/Room_Type_A.java
 

public class Room_Type_A extends Room{
	protected double price_per_day;//timh ana mera
	//constructors
	public Room_Type_A()
	{
	    
	}
	public Room_Type_A(int maximum_capacity,double price_per_person,double price){
	    super(maximum_capacity,price_per_person);//kanei gia thn megisth xwrithkothta kai gia thn timh ana atomo 
	    //oti kai o constructor ths Room
	   this.price_per_day=price;//thetei thn thmh an hmera ish me thn timh sto orisma
	   }
	//methods
	double pricing()// esoda apo to dwmatio
	{
		double price=0;//arxikopoihsh esodvn sto 0
		int f=0;//metrhths hmerwn me krathsh arxikopoieitai sto 0
		for (int c=0; c<availability.length; c++)//diatrexei ton pinaka diathesimothtas
		{
			if(availability[c]!=null)//an to keli tou pinaka den einai keno
			{
				f++;//auksanei o metrhths  hmerwn
			}
			price=price_per_day*f;//esoda = timh ana hmera epi tis hmeres me krathsh sto dwmatio
		}
		return price;//epistrefei ta esoda
		}
	public double getPrice_per_day() 
	{
		return price_per_day;
	}

	public void setPrice_per_day(int price_per_day) {
		this.price_per_day = price_per_day;
}
	}

File: 1st year/hotel project c++/projectcpp/src/Room_Type_A.cpp
#include "Room_Type_A.h"
#include"Room.h"
Room_Type_A::Room_Type_A()
{
    //ctor
}

Room_Type_A::~Room_Type_A()
{
    //dtor
}
Room_Type_A:: Room_Type_A(int maximum_capacity, double price_per_person, double price): Room(maximum_capacity, price_per_person)
{
	//kanei gia thn megisth xwrithkothta kai gia thn timh ana atomo oti kai o constructor ths Room
	price_per_day = price;//thetei thn thmh an hmera ish me thn timh sto orisma
}
//methods
double Room_Type_A:: pricing()// esoda apo to dwmatio
{
	double price = 0;//arxikopoihsh esodvn sto 0
	int f = 0;//metrhths hmerwn me krathsh arxikopoieitai sto 0
	for (int i = 1; i < N; i++)//diatrexei ton pinaka diathesimothtas
	{
		if (availability[i] != 0)//an to keli tou pinaka den einai keno
		{
			f++;//auksanei o metrhths  hmerwn
		}
		price = price_per_day*f;//esoda = timh ana hmera epi tis hmeres me krathsh sto dwmatio
	}
	return price;//epistrefei ta esoda
	}
